packages/mermaid/src/diagrams/quadrant-chart/quadrantBuilder.ts

```diff
@@ -0,0 +1,385 @@
+// @ts-ignore: TODO Fix ts errors
+import { scaleLinear } from 'd3';
+
+export type QuadrantPointInputType = { x: number; y: number; text: string };
+
+export type TextVerticalPos = 'left' | 'center' | 'right';
+export type TextHorizontalPos = 'top' | 'middle' | 'bottom';
+
+export type QuadrantTextType = {
+  text: string;
+  fill: string;
+  x: number;
+  y: number;
+  verticalPos: TextVerticalPos;
+  horizontalPos: TextHorizontalPos;
+  fontSize: number;
+  rotation: number;
+};
+
+export type QuadrantPointType = {
+  x: number;
+  y: number;
+  fill: string;
+  radius: number;
+  text: QuadrantTextType;
+};
+
+export type QuadrantQuadrantsType = {
+  text: QuadrantTextType;
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+  fill: string;
+};
+
+export type QuadrantBuildType = {
+  points: QuadrantPointType[];
+  quadrants: QuadrantQuadrantsType[];
+  axisLabels: QuadrantTextType[];
+};
+
+export class QuadrantBuilder {
+  private _quadrant1Text = '';
+  private _quadrant2Text = '';
+  private _quadrant3Text = '';
+  private _quadrant4Text = '';
+  private _xAxisLeftText = '';
+  private _xAxisRightText = '';
+  private _yAxisBottomText = '';
+  private _yAxisTopText = '';
+  private _totalHeight = 500;
+  private _totalWidth = 500;
```

Comment:
Why are these private with getters and setters (which don't do any validation or processing)?
Would be cleaner to make it public no?

Reply:
I thought of doing the sanitize on the builder but later realized it should be in the db, changed it.